core/src/scene/sceneLoader.cpp

```diff
@@ -954,29 +975,44 @@ void SceneLoader::loadSource(const std::shared_ptr<Platform>& platform, const st
     }
 
     // distinguish tiled and non-tiled sources by url
-    bool tiled = url.find("{x}") != std::string::npos &&
+    bool tiled = url.size() > 0 &&
+        url.find("{x}") != std::string::npos &&
         url.find("{y}") != std::string::npos &&
         url.find("{z}") != std::string::npos;
 
+    bool isMBTilesFile = false;
+    {
+        const char* extStr = ".mbtiles";
+        const size_t extLength = strlen(extStr);
+        const size_t urlLength = url.length();
+        isMBTilesFile = urlLength > extLength && (url.compare(urlLength - extLength, urlLength, extStr) == 0);
```

Comment:
Right - your comment about `tiled`. In the case of something like `url: /w/{x}/{y}/{z}.mbtiles` I would lean towards treating it as a path to an MBTiles file - curly braces are valid in paths and are somewhat common on Windows. I would not make it a recommended practice though ;)